app/models/spree/product.rb

```diff
@@ -109,7 +109,7 @@ class Product < ApplicationRecord
                              if: proc { Spree::Config[:products_require_tax_category] }
 
     validates :variant_unit, presence: true
-    validates :unit_value, presence: { if: ->(p) { %w(weight volume).include? p.variant_unit } }
+    validates :unit_value, presence: true
```

Comment:
This is not correct. You are changing the validation logic here and this is what is breaking specs. The issue you are working on is only about the feedback the user gets when trying to create a product. It's not about the internal logic.